homework/src/index.js

```diff
@@ -1,47 +1,119 @@
+              : `Updated: ${new Date(changeEvent.updated_at).toLocaleString()}`,
+          class: "updated"
+        });
+        sectLinfobox = crnap("div", sectLinfobox, {
+          html: null == changeEvent.size ? "No size" : `Size: ${changeEvent.size}`,
+          class: "size"
+        });
+        fetchJSON(changeEvent.contributors_url, (err, contributors) => {
+          let fetchToDo = contributors.length;
+          contributors.forEach(contributor => {
+            fetchJSON(contributor.url, (err, user) => {
+              Object.assign(contributor, user);
+              --fetchToDo;
+              0 === fetchToDo &&
+                (console.log(contributors),
+                  contributors.forEach(contrUser => {
+                    let sectRcontrs = crnap("div", sectR, { id: "contributors" });
+                    sectRcontrs = crnap("div", sectRcontrs, { class: "card" });
+                    let secRimglink = crnap("a", sectRcontrs, { href: contrUser.html_url });
+                    secRimglink = crnap("figure", secRimglink, {});
+                    crnap("img", secRimglink, { src: contrUser.avatar_url });
+                    crnap("h2", sectRcontrs, { html: null == contrUser.name ? "" : `name: ${contrUser.name}` });
+                    crnap("p", sectRcontrs, { html: null == contrUser.login ? "" : `nickname: ${contrUser.login}` });
+                    crnap("p", sectRcontrs, {
+                      html: null == contrUser.location ? "" : `location: ${contrUser.location}`
+                    });
+                    crnap("p", sectRcontrs, {
+                      html: null == contrUser.followers ? "" : `followers: ${contrUser.followers}`
+                    });
+                    crnap("p", sectRcontrs, {
+                      html: null == contrUser.company ? "" : `company: ${contrUser.company}`
+                    });
+                    crnap("p", sectRcontrs, {
+                      html: null == contrUser.contributions ? "" : `contributions: ${contrUser.contributions}`
+                    });
+                    crnap("p", sectRcontrs, {
+                      html:
+                        null == contrUser.updated_at
+                          ? ""
+                          : `last github visit: ${new Date(contrUser.updated_at).toLocaleString()}`
+                    });
+                  }));
+            });
+          });
+        });
       }
-    };
-    xhr.onerror = () => cb(new Error('Network request failed'));
-    xhr.send();
+      const root = document.getElementById("root");
+      let header = crnap("header", root, {});
+      header = crnap("div", header, { class: "selectfield" });
+      const menu = crnap("select", header, { id: "selectmenu" });
+      var main = crnap("main", root, { id: "main" });
+      err
+        ? crnap("div", root, { a: err.message, b: "alert-error" })
+        : (data
+          .sort((b, a) => b.name.localeCompare(a.name))
```

Comment:
While you can use the ternary operator like this, it is not very readable code. An `if-then-else` would more clearly express your intention:

```js
if (err) {
  //...
} else {
  //..
}

Additionally, your error `div` is not rendered visually. It falls off the screen because your `<main>` element has a 100% height. Did you check the error handling yourself?